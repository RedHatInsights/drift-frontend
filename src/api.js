/*eslint-disable camelcase*/
import axios from 'axios';
import { DRIFT_API_ROOT, BASELINE_API_ROOT, HISTORICAL_PROFILES_API_ROOT } from './constants';

async function post(path, body = {}) {
    const request = await axios.post(DRIFT_API_ROOT.concat(path), body);
    return request.data;
}

async function getBaselines(path, getParams = {}) {
    const request = await axios.get(BASELINE_API_ROOT.concat(path), { params: getParams });
    return request.data;
}

async function getBaseline(path) {
    const request = await axios.get(BASELINE_API_ROOT.concat(path));
    return request.data.data[0];
}

async function getNotificationSystems(path) {
    const request = await axios.get(BASELINE_API_ROOT.concat(path));
    return request.data.system_ids;
}

async function postSystemNotifications(path, body = {}) {
    const request = await axios.post(BASELINE_API_ROOT.concat(path), body);
    return request.data;
}

async function patchBaseline(path, body = {}) {
    let request = await axios.patch(BASELINE_API_ROOT.concat(path), body);

    return request;
}

async function postBaseline(path, body = {}) {
    const request = await axios.post(BASELINE_API_ROOT.concat(path), body);
    return request.data;
}

async function deleteBaselines(path, body = {}) {
    let response = await axios.post(BASELINE_API_ROOT.concat(path), body);

    return response;
}

async function getHistoricalData(path) {
    let response;

    const request = await axios.get(HISTORICAL_PROFILES_API_ROOT.concat(path))
    .catch(function (error) {
        return error.response;
    });

    if (request.status === 200) {
        response = request.data.data[0];
    } else {
        response = request;
    }

    return response;
}

function getCompare(systemIds = [], baselineIds = [], historicalSystemProfileIds = [], referenceId = '') {
    if (!Array.isArray(systemIds)) {
        systemIds = [ systemIds ];
    }

    if (!Array.isArray(baselineIds)) {
        baselineIds = [ baselineIds ];
    }

    if (!Array.isArray(historicalSystemProfileIds)) {
        historicalSystemProfileIds = [ historicalSystemProfileIds ];
    }

    return post('/comparison_report', {
        system_ids: systemIds,
        baseline_ids: baselineIds,
        historical_system_profile_ids: historicalSystemProfileIds,
        reference_id: referenceId
    });
}

function getBaselineList(params = {}) {
    return getBaselines('/baselines', params);
}

function getBaselineData(baselineId = []) {
    let path = '/baselines/';

    return getBaseline(path.concat(baselineId), {});
}

function patchBaselineData(baselineId = [], apiBody) {
    let path = '/baselines/';

    return patchBaseline(path.concat(baselineId), apiBody);
}

function deleteBaselinesData(deleteBaselinesAPIBody = []) {
    let path = '/baselines/deletion_request';

    return deleteBaselines(path, deleteBaselinesAPIBody);
}

function postNewBaseline(apiBody, uuid) {
    let path = '/baselines';
    if (uuid !== undefined) {
        let newPath = path.concat('/', uuid, '?', 'display_name=', apiBody.display_name);
        return postBaseline(newPath, {});
    }

    return postBaseline(path, apiBody);
}

function fetchHistoricalData(systemId) {
    let path = '/systems/';
    return getHistoricalData(path.concat(systemId));
}

function getBaselineNotification(baselineId) {
    let path = `/baselines/${baselineId}/systems`;
    return getNotificationSystems(path);
}

function addSystemNotification(baselineId, systems) {
    let path = `/baselines/${baselineId}/systems`;
    let body = { system_ids: systems };

    return postSystemNotifications(path, body);
}

function deleteSystemNotifications(baselineId, systems) {
    let path = `/baselines/${baselineId}/systems/deletion_request`;
    let body = { system_ids: systems };

    return postSystemNotifications(path, body);
}

export default {
    getCompare,
    getBaselineList,
    getBaselineData,
    patchBaselineData,
    deleteBaselinesData,
    postNewBaseline,
    fetchHistoricalData,
    getBaselineNotification,
    addSystemNotification,
    deleteSystemNotifications
};
/*eslint-enable camelcase*/
