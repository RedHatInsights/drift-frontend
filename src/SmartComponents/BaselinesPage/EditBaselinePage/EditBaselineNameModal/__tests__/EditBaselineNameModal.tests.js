import React from 'react';

import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import toJson from 'enzyme-to-json';

import ConnectedEditBaselineNameModal, { EditBaselineNameModal } from '../EditBaselineNameModal';
import editBaselineFixtures from '../../EditBaseline/__tests__/helpers.fixtures';
import api from '../../../../../api';

describe('EditBaselineNameModal', () => {
    let props;

    beforeEach(() => {
        props = {
            baselineData: editBaselineFixtures.mockBaselineData1,
            modalOpened: true,
            error: {}
        };
    });

    it('should render correctly', () => {
        const wrapper = shallow(
            <EditBaselineNameModal { ...props }/>
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render mount correctly', () => {
        const wrapper = mount(
            <EditBaselineNameModal { ...props }/>
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    describe('API', () => {
        it('should update baseline name', () => {
            const wrapper = shallow(
                <EditBaselineNameModal { ...props }/>
            );

            wrapper.find('TextInput').simulate('change', 'new-lotr-baseline');
            expect(wrapper.state('baselineName')).toEqual('new-lotr-baseline');
        });

        it('should confirm modal with patch', () => {
            const patchBaseline = jest.fn();
            const wrapper = mount(
                <EditBaselineNameModal
                    { ...props }
                    patchBaseline={ patchBaseline }
                />
            );

            wrapper.find('.pf-c-button').at(1).simulate('click');
            expect(patchBaseline).toHaveBeenCalled();
        });

        it('should confirm modal with error', () => {
            props.error = { detail: 'this is an error', status: 400 };
            const wrapper = mount(
                <EditBaselineNameModal
                    { ...props }
                />
            );

            wrapper.find('.pf-c-button').at(1).simulate('click');
            expect(wrapper.find('FormGroup').prop('helperTextInvalid')).toEqual('this is an error');
        });

        it('should cancel modal', () => {
            const toggleEditNameModal = jest.fn();
            const wrapper = mount(
                <EditBaselineNameModal
                    { ...props }
                    toggleEditNameModal={ toggleEditNameModal }
                />
            );

            wrapper.find('.pf-c-button').at(0).simulate('click');
            expect(toggleEditNameModal).toHaveBeenCalled();
        });

        it('should catch key press', () => {
            const patchBaseline = jest.fn();
            const wrapper = shallow(
                <EditBaselineNameModal
                    { ...props }
                    patchBaseline={ patchBaseline }
                />
            );

            wrapper.find('FormGroup').simulate('keypress', { key: 'Enter', preventDefault: jest.fn() });
            expect(patchBaseline).toHaveBeenCalled();
        });
    });

    describe('ConnectedEditBaselineNameModal', () => {
        let initialState;
        let mockStore;
        let props;

        beforeEach(() => {
            mockStore = configureStore();
            initialState = {};
            props = {
                baselineData: editBaselineFixtures.mockBaselineData1,
                modalOpened: true,
                error: {}
            };
        });

        it('should render correctly', () => {
            const store = mockStore(initialState);
            const patchBaseline = jest.fn();

            const wrapper = mount(
                <MemoryRouter keyLength={ 0 }>
                    <Provider store={ store }>
                        <ConnectedEditBaselineNameModal
                            { ...props }
                            patchBaseline={ patchBaseline }
                        />
                    </Provider>
                </MemoryRouter>
            );

            const actions = store.getActions();
            wrapper.find('.pf-c-button').at(1).simulate('click');
            expect(actions).toEqual([{ type: 'PATCH_BASELINE', payload: api.patchBaselineData('blah') }]);
        });
    });
});
