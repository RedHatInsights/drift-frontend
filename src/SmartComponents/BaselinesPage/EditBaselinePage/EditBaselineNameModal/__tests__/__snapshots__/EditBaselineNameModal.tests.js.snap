// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditBaselineNameModal should render correctly 1`] = `
<Modal
  actions={
    Array [
      <Button
        onClick={[Function]}
        ouiaId="save"
        variant="primary"
      >
        Save
      </Button>,
      <Button
        onClick={[Function]}
        ouiaId="cancel"
        variant="link"
      >
        Cancel
      </Button>,
    ]
  }
  appendTo={[Function]}
  aria-describedby=""
  aria-label=""
  aria-labelledby=""
  className="drift"
  hasNoBodyWrapper={false}
  isOpen={true}
  onClose={[Function]}
  ouiaSafe={true}
  showClose={true}
  title="Edit name"
  titleIconVariant={null}
  titleLabel=""
  variant="small"
>
  <div
    className="fact-value"
  >
    <Form>
      <FormGroup
        fieldId="baseline name"
        helperTextInvalid={null}
        isRequired={true}
        label="Baseline title"
        onKeyPress={[Function]}
        validated={null}
      >
        <TextInput
          aria-label="baseline name"
          onChange={[Function]}
          type="text"
          validated={null}
          value="lotr-baseline"
        />
      </FormGroup>
    </Form>
  </div>
</Modal>
`;

exports[`EditBaselineNameModal should render mount correctly 1`] = `
<EditBaselineNameModal
  baselineData={
    Object {
      "account": "12345678",
      "baseline_facts": Array [
        Object {
          "name": "Sauron",
          "value": "the Dark Lord",
        },
        Object {
          "name": "Galadriel",
          "value": "the Elven Queen",
        },
        Object {
          "name": "The Fellowship of the Ring",
          "values": Array [
            Object {
              "name": "Frodo",
              "value": "Baggins",
            },
            Object {
              "name": "Samwise",
              "value": "Gamgee",
            },
            Object {
              "name": "Gandalf",
              "value": "the Grey",
            },
            Object {
              "name": "Meriadoc",
              "value": "Brandybuck",
            },
            Object {
              "name": "Peregrin",
              "value": "Took",
            },
            Object {
              "name": "Gimli",
              "value": "son of Gloin",
            },
            Object {
              "name": "Legolas",
              "value": "Greenleaf",
            },
            Object {
              "name": "Boromir",
              "value": "son of Denethor",
            },
            Object {
              "name": "Aragorn",
              "value": "son of Arathorn",
            },
          ],
        },
      ],
      "created": "2019-10-17T16:20:06.050710Z",
      "display_name": "lotr-baseline",
      "fact_count": 3,
      "id": "jf8alskj-jf74-aje8-ke83-jf84ldjru439k",
      "updated": "2020-02-06T21:48:30.661510Z",
    }
  }
  error={Object {}}
  modalOpened={true}
>
  <Modal
    actions={
      Array [
        <Button
          onClick={[Function]}
          ouiaId="save"
          variant="primary"
        >
          Save
        </Button>,
        <Button
          onClick={[Function]}
          ouiaId="cancel"
          variant="link"
        >
          Cancel
        </Button>,
      ]
    }
    appendTo={[Function]}
    aria-describedby=""
    aria-label=""
    aria-labelledby=""
    className="drift"
    hasNoBodyWrapper={false}
    isOpen={true}
    onClose={[Function]}
    ouiaSafe={true}
    showClose={true}
    title="Edit name"
    titleIconVariant={null}
    titleLabel=""
    variant="small"
  >
    <Portal
      containerInfo={
        <div>
          <div
            class="pf-c-backdrop"
          >
            <div
              class="pf-l-bullseye"
            >
              <div
                aria-describedby="pf-modal-part-2"
                aria-labelledby="pf-modal-part-1"
                aria-modal="true"
                class="pf-c-modal-box drift pf-m-sm"
                data-ouia-component-id="OUIA-Generated-Modal-small-1"
                data-ouia-component-type="PF4/ModalContent"
                data-ouia-safe="true"
                id="pf-modal-part-0"
                role="dialog"
              >
                <button
                  aria-disabled="false"
                  aria-label="Close"
                  class="pf-c-button pf-m-plain"
                  data-ouia-component-id="OUIA-Generated-Button-plain-1"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 352 512"
                    width="1em"
                  >
                    <path
                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                    />
                  </svg>
                </button>
                <header
                  class="pf-c-modal-box__header"
                >
                  <h1
                    class="pf-c-modal-box__title"
                    id="pf-modal-part-1"
                  >
                    
                    <span
                      class="pf-c-modal-box__title-text"
                    >
                      Edit name
                    </span>
                  </h1>
                </header>
                <div
                  class="pf-c-modal-box__body"
                  id="pf-modal-part-2"
                >
                  <div
                    class="fact-value"
                  >
                    <form
                      class="pf-c-form"
                      novalidate=""
                    >
                      <div
                        class="pf-c-form__group"
                      >
                        <div
                          class="pf-c-form__group-label"
                        >
                          <label
                            class="pf-c-form__label"
                            for="baseline name"
                          >
                            <span
                              class="pf-c-form__label-text"
                            >
                              Baseline title
                            </span>
                            <span
                              aria-hidden="true"
                              class="pf-c-form__label-required"
                            >
                               
                              *
                            </span>
                          </label>
                           
                        </div>
                        <div
                          class="pf-c-form__group-control"
                        >
                          <input
                            aria-invalid="false"
                            aria-label="baseline name"
                            class="pf-c-form-control"
                            type="text"
                            value="lotr-baseline"
                          />
                          
                        </div>
                      </div>
                    </form>
                  </div>
                </div>
                <footer
                  class="pf-c-modal-box__footer"
                >
                  <button
                    aria-disabled="false"
                    class="pf-c-button pf-m-primary"
                    data-ouia-component-id="save"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    type="button"
                  >
                    Save
                  </button>
                  <button
                    aria-disabled="false"
                    class="pf-c-button pf-m-link"
                    data-ouia-component-id="cancel"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    type="button"
                  >
                    Cancel
                  </button>
                </footer>
              </div>
            </div>
          </div>
        </div>
      }
    >
      <ModalContent
        actions={
          Array [
            <Button
              onClick={[Function]}
              ouiaId="save"
              variant="primary"
            >
              Save
            </Button>,
            <Button
              onClick={[Function]}
              ouiaId="cancel"
              variant="link"
            >
              Cancel
            </Button>,
          ]
        }
        aria-describedby=""
        aria-label=""
        aria-labelledby=""
        boxId="pf-modal-part-0"
        className="drift"
        descriptorId="pf-modal-part-2"
        hasNoBodyWrapper={false}
        isOpen={true}
        labelId="pf-modal-part-1"
        onClose={[Function]}
        ouiaId="OUIA-Generated-Modal-small-1"
        ouiaSafe={true}
        showClose={true}
        title="Edit name"
        titleIconVariant={null}
        titleLabel=""
        variant="small"
      >
        <Backdrop>
          <div
            className="pf-c-backdrop"
          >
            <FocusTrap
              active={true}
              className="pf-l-bullseye"
              focusTrapOptions={
                Object {
                  "clickOutsideDeactivates": true,
                }
              }
              paused={false}
            >
              <div
                className="pf-l-bullseye"
              >
                <ModalBox
                  aria-describedby="pf-modal-part-2"
                  aria-label=""
                  aria-labelledby="pf-modal-part-1"
                  className="drift"
                  data-ouia-component-id="OUIA-Generated-Modal-small-1"
                  data-ouia-component-type="PF4/ModalContent"
                  data-ouia-safe={true}
                  id="pf-modal-part-0"
                  style={Object {}}
                  variant="small"
                >
                  <div
                    aria-describedby="pf-modal-part-2"
                    aria-label={null}
                    aria-labelledby="pf-modal-part-1"
                    aria-modal="true"
                    className="pf-c-modal-box drift pf-m-sm"
                    data-ouia-component-id="OUIA-Generated-Modal-small-1"
                    data-ouia-component-type="PF4/ModalContent"
                    data-ouia-safe={true}
                    id="pf-modal-part-0"
                    role="dialog"
                    style={Object {}}
                  >
                    <ModalBoxCloseButton
                      onClose={[Function]}
                    >
                      <Button
                        aria-label="Close"
                        className=""
                        onClick={[Function]}
                        variant="plain"
                      >
                        <button
                          aria-disabled={false}
                          aria-label="Close"
                          className="pf-c-button pf-m-plain"
                          data-ouia-component-id="OUIA-Generated-Button-plain-1"
                          data-ouia-component-type="PF4/Button"
                          data-ouia-safe={true}
                          disabled={false}
                          onClick={[Function]}
                          role={null}
                          type="button"
                        >
                          <TimesIcon
                            color="currentColor"
                            noVerticalAlign={false}
                            size="sm"
                          >
                            <svg
                              aria-hidden={true}
                              aria-labelledby={null}
                              fill="currentColor"
                              height="1em"
                              role="img"
                              style={
                                Object {
                                  "verticalAlign": "-0.125em",
                                }
                              }
                              viewBox="0 0 352 512"
                              width="1em"
                            >
                              <path
                                d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                              />
                            </svg>
                          </TimesIcon>
                        </button>
                      </Button>
                    </ModalBoxCloseButton>
                    <ModalBoxHeader
                      help={null}
                    >
                      <header
                        className="pf-c-modal-box__header"
                      >
                        <ModalBoxTitle
                          id="pf-modal-part-1"
                          title="Edit name"
                          titleIconVariant={null}
                          titleLabel=""
                        >
                          <h1
                            className="pf-c-modal-box__title"
                            id="pf-modal-part-1"
                          >
                            <span
                              className="pf-c-modal-box__title-text"
                            >
                              Edit name
                            </span>
                          </h1>
                        </ModalBoxTitle>
                      </header>
                    </ModalBoxHeader>
                    <ModalBoxBody
                      id="pf-modal-part-2"
                    >
                      <div
                        className="pf-c-modal-box__body"
                        id="pf-modal-part-2"
                      >
                        <div
                          className="fact-value"
                        >
                          <Form>
                            <form
                              className="pf-c-form"
                              noValidate={true}
                            >
                              <FormGroup
                                fieldId="baseline name"
                                helperTextInvalid={null}
                                isRequired={true}
                                label="Baseline title"
                                onKeyPress={[Function]}
                                validated={null}
                              >
                                <div
                                  className="pf-c-form__group"
                                  onKeyPress={[Function]}
                                >
                                  <div
                                    className="pf-c-form__group-label"
                                  >
                                    <label
                                      className="pf-c-form__label"
                                      htmlFor="baseline name"
                                    >
                                      <span
                                        className="pf-c-form__label-text"
                                      >
                                        Baseline title
                                      </span>
                                      <span
                                        aria-hidden="true"
                                        className="pf-c-form__label-required"
                                      >
                                         
                                        *
                                      </span>
                                    </label>
                                     
                                  </div>
                                  <div
                                    className="pf-c-form__group-control"
                                  >
                                    <TextInput
                                      aria-label="baseline name"
                                      onChange={[Function]}
                                      type="text"
                                      validated={null}
                                      value="lotr-baseline"
                                    >
                                      <TextInputBase
                                        aria-label="baseline name"
                                        className=""
                                        innerRef={null}
                                        isDisabled={false}
                                        isLeftTruncated={false}
                                        isReadOnly={false}
                                        isRequired={false}
                                        onChange={[Function]}
                                        type="text"
                                        validated={null}
                                        value="lotr-baseline"
                                      >
                                        <input
                                          aria-invalid={false}
                                          aria-label="baseline name"
                                          className="pf-c-form-control"
                                          disabled={false}
                                          onBlur={[Function]}
                                          onChange={[Function]}
                                          onFocus={[Function]}
                                          readOnly={false}
                                          required={false}
                                          type="text"
                                          value="lotr-baseline"
                                        />
                                      </TextInputBase>
                                    </TextInput>
                                  </div>
                                </div>
                              </FormGroup>
                            </form>
                          </Form>
                        </div>
                      </div>
                    </ModalBoxBody>
                    <ModalBoxFooter>
                      <footer
                        className="pf-c-modal-box__footer"
                      >
                        <Button
                          key="confirm"
                          onClick={[Function]}
                          ouiaId="save"
                          variant="primary"
                        >
                          <button
                            aria-disabled={false}
                            aria-label={null}
                            className="pf-c-button pf-m-primary"
                            data-ouia-component-id="save"
                            data-ouia-component-type="PF4/Button"
                            data-ouia-safe={true}
                            disabled={false}
                            onClick={[Function]}
                            role={null}
                            type="button"
                          >
                            Save
                          </button>
                        </Button>
                        <Button
                          key="cancel"
                          onClick={[Function]}
                          ouiaId="cancel"
                          variant="link"
                        >
                          <button
                            aria-disabled={false}
                            aria-label={null}
                            className="pf-c-button pf-m-link"
                            data-ouia-component-id="cancel"
                            data-ouia-component-type="PF4/Button"
                            data-ouia-safe={true}
                            disabled={false}
                            onClick={[Function]}
                            role={null}
                            type="button"
                          >
                            Cancel
                          </button>
                        </Button>
                      </footer>
                    </ModalBoxFooter>
                  </div>
                </ModalBox>
              </div>
            </FocusTrap>
          </div>
        </Backdrop>
      </ModalContent>
    </Portal>
  </Modal>
</EditBaselineNameModal>
`;
